utils.js

```diff
@@ -187,8 +187,27 @@ function dateFromShortDate(shortdate) {
                              0, 0, 0 );
 }
 
-function getImageList(settings) {
-    return JSON.parse(settings.get_string('bing-json'));
+function getImageList(settings, filter = null) {
+    let image_list = JSON.parse(settings.get_string('bing-json'));
+    if (!filter) {
+        return image_list;
+    }
+    else {
+        let filtered = [];
+        image_list.forEach(function(x, i) {
+            let ignore = false;
+            if (filter.faves && !x.favourite)
+                ignore = true;
+            if (filter.min_height && x.height < filter.min_height)
+                ignore = true;
+            if (filter.hidden && x.hidden)
+                ignore = true;
+            
+            if (!ignore)
+                filtered.push(x);
+        });
+        return filtered;
```

Comment:
This could use the filter function directly instead of forEach + push

Reply:
Yeah there is a bit of refactoring I need to do for the awkward logic that's kind of been hacked together over time. This one is a new hack though, haha.

Reply:
Done, thanks for review @mercutiodesign